ui/components/app/permission-page-container/permission-page-container.component.js

```diff
@@ -146,10 +146,11 @@ export default class PermissionPageContainer extends Component {
           (selectedAccount) => selectedAccount.address,
         ),
       }),
-      ...(_request.permissions.permittedChains && {
-        approvedChainIds: _request.permissions?.permittedChains?.caveats.find(
-          (caveat) => caveat.type === 'restrictNetworkSwitching',
-        )?.value,
+      ...(_request.permissions[PermissionNames.permittedChains] && {
```

Comment:
I think for the sake of readability we could split this one-liner logic into several lines of code with named variables

Reply:
good point. done here https://github.com/MetaMask/metamask-extension/pull/26635/commits/0dc84aafa710174fa4ad9569a18bf965ce3e45e7